### DISCUSSION TITLE
Need help formulating this command

### DISCUSSION THREAD

**Discussion by SentientBeingOnAQuest** [ImageMagick/ImageMagick#4578] [Help]
I'm a total noob when it comes to ImageMagick but I have received some help from others in formulating the following commands:

magick image.jpg +write mpr:img -blur 40x10 -resize 600x600! mpr:img -gravity center -compose over -composite result.jpg

magick image.jpg -bordercolor LightGrey -border 20x10 result2.jpg

I would like to combine these commands and create a for loop which applies them to all of the jpgs in a given folder.
Can anyone help me with this? 

I'm on imagemagick 7 for windows 10 and I'm using the default vanilla windows command window.

This is what I'm trying to achieve:

The first command does this:

![result](https://user-images.githubusercontent.com/95970582/145683739-97b268f9-1187-4926-8574-51a9163a1a56.jpg)

The second command does this:

![result2](https://user-images.githubusercontent.com/95970582/145683740-1fdfe881-e509-4b50-9079-a14f2ebe995e.jpg)

What I'm trying to achieve is this:

![result desired](https://user-images.githubusercontent.com/95970582/145683738-f8990634-335e-49c4-a277-2d5bdabc2116.jpg)

**Comment by snibgo** [comment]
You could combine into one command them like this (untested). Windows syntax
```
magick ^
  image.jpg +write mpr:img ^
  -blur 40x10 -resize 600x600! ^
  mpr:img -gravity center -compose over -composite ^
  -write result.jpg ^
  +delete ^
  mpr:img -bordercolor LightGrey -border 20x10 result2.jpg
```
I have split it into lines for clarity.

To apply this to all the jpgs in the current folder, put it inside a "for" loop, like this (also untested):
```
for %F in (*.jpg) do magick ^
  %F +write mpr:img ^
  -blur 40x10 -resize 600x600! ^
  mpr:img -gravity center -compose over -composite ^
  -write result.jpg ^
  +delete ^
  mpr:img -bordercolor LightGrey -border 20x10 result2.jpg
```
If you put this in a BAT script, double each %, so "%%F".

This overwrites the output files for all the inputs, so you might want different output filenames. Also writing the outputs to a different directory would be wise, so re-running the script doesn't also process all the new files.

**Reply by SentientBeingOnAQuest** [reply]
Thanks for your answer,
Running your first command actually produces 2 separate images with only 1 part of the command being applied to each.
One of which has the desired background, and the other has the desired border around the foreground image. I would like the two actions to be combined into one image.

On the second command the for loop part works very well, the command window shows every file being processed, however I am getting the following errors:

magick: unable to open image 'maxresdefault': No such file or directory @ error/blob.c/OpenBlob/3527.
magick: no decode delegate for this image format `' @ error/constitute.c/ReadImage/572.

**Reply by SentientBeingOnAQuest** [reply]
PS I have updated my original post to give more clarity on what I'm trying to achieve

**Comment by snibgo** [comment]
You haven't shown your input image.
